### PR TITLE
Add changelog and update readme

### PR DESCRIPTION
**Pull request by yunchu**
### Summary

* add CHANGELOG.md
* fix utf-8 encoding issue
* update badges in README.md

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sungmanc — DISMISSED** [review]

**Review by harimkang — DISMISSED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -11,12 +11,12 @@ build-backend = "setuptools.build_meta"
 [project]
 name = "otx"
 dynamic = ["version"]
-description = "OpenVINO™ Training Extensions: Train, Evaluate, Optimize, Deploy Computer Vision Models via OpenVINO™"
+description = "OpenVINO Training Extensions: Train, Evaluate, Optimize, Deploy Computer Vision Models via OpenVINO"
```
Comment (vinnamkim) [inline pyproject.toml]: Why is the trademark character removed?

Comment (yunchu) [reply pyproject.toml]: @vinnamkim from some dev environment (e.g., WSL), pre-commit (ruff) complained about this file while decoding to utf-8. it seems to be passed in my current environment but i think TM is not important thing to the project description so i've removed it. is there any concerns about it?

Comment (vinnamkim) [reply pyproject.toml]: image
The trademark symbol is definitely present at least in sources provided by Intel. I don't think it is good idea to drop it.

Comment (yunchu) [reply pyproject.toml]: This is the error that i've met before - https://github.com/openvinotoolkit/training_extensions/actions/runs/8277980414/job/22649413049
Anyway, I will try to check again with adding ™ to the project description.

**Review by vinnamkim — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Pull request merged**